crates/rover-client/src/error.rs

```diff
@@ -0,0 +1,30 @@
+use thiserror::Error;
+
+/// An enum represent all type of possible failures.
+#[derive(Error, Debug)]
+pub enum RoverClientError {
+    /// The provided GraphQL was invalid.
+    #[error("encountered a GraphQL error, registry responded with: {msg}")]
+    GraphQLError {
+        /// The error message.
+        msg: String,
+    },
+    /// Tried to build a [HeaderMap] with an invalid header value.
+    #[error("header value was not valid: {msg:?}")]
+    HeadersError {
+        /// The error message.
+        msg: http::header::InvalidHeaderValue,
+    },
+    /// Encountered an error handling the received response.
+    #[error("encountered an error handling the response: {msg}")]
+    ResponseError {
```

Comment:
```suggestion
    HandleResponse {
```

Reply:
so if you type in the wrong graph name, you get this error, which would be nice to clean up a bit:

`encountered an error handling the response: No service found`

if it's the wrong name, the error message could be like `invalid graph name, please enter a valid graph title, which you can find at {insert link}`